src/core/src/bound_evaluate.cpp

```diff
@@ -172,13 +171,10 @@ bool default_bound_evaluator(const ov::Node* node,
     return node->evaluate(output_values, inputs);
 }
 
-ov::Tensor equality_mask(const ov::Tensor& tensor, const std::shared_ptr<op::v0::Constant>& constant) {
+ov::Tensor equality_mask(const ov::Tensor& tensor, const ov::Tensor& c_tensor) {
     auto mask_out = ov::TensorVector{{element::boolean, tensor.get_shape()}};
-
-    auto c_tensor = ov::Tensor(constant->get_element_type(), constant->get_shape());
-    memcpy(c_tensor.data(), constant->get_data_ptr(), c_tensor.get_byte_size());
-
     const auto& param = std::make_shared<op::v0::Parameter>(tensor.get_element_type(), tensor.get_shape());
+    const auto& constant = std::make_shared<op::v0::Parameter>(c_tensor.get_element_type(), c_tensor.get_shape());
```

Comment:
Is it used anywhere and why parameter is named as constant?

Reply:
It's used line below for equality evaluation. The name is a bit unfortunate now, I'll change it.